### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by exactly-one-kas**

### CODE REVIEW

**Comment by exactly-one-kas** [comment]
```
100% tests passed, 0 tests failed out of 15
Total Test time (real) = 157.91 sec
```

File: src/ui/DiffTreeModel.cpp [hunk 1]
```diff
@@ -282,10 +282,10 @@ bool DiffTreeModel::discard(const QModelIndex &index)
 				submodule.update(nullptr, false, true); // In Repo view it is done asynchron. Maybe changing here too!
 				break;
 			}
-			if (!is_submodule)
-				filePatches.append(trackedPatch);
-
```
Comment (Murmele) [inline src/ui/DiffTreeModel.cpp:287]: Thanks! I inserted into the wrong bracket

File: src/ui/TreeView.h [hunk 2]
```diff
@@ -26,6 +26,7 @@ class TreeView : public QTreeView
   void onCustomContextMenu(const QPointF& point);
   bool eventFilter(QObject *obj, QEvent *event) override;
   void deselectAll();
+  QRect checkRect(const QModelIndex &index);
```
Comment (Murmele) [inline src/ui/TreeView.h:29]:
```suggestion
  /*
   * Simulate checking an index. Used in the Unit tests
   */
  QRect checkRect(const QModelIndex &index);
```

**Comment by exactly-one-kas** [comment]
For some reason, the tests fail in GitHub Actions, but work fine on my machine.

Until actions/runner#1206 gets resolved I'm unable to properly investigate why.

If someone else can reproduce this on their machine and wants to continue this, please do

**Comment by Murmele** [comment]
I will check why flatpak fails

**Pull request merged**